docs/security-considerations.rst

```diff
@@ -246,21 +246,31 @@ Two's Complement / Underflows / Overflows
 =========================================
 
 As in many programming languages, Solidity's integer types are not actually integers.
-They resemble integers when the values are small, but behave differently if the numbers are larger.
-For example, the following is true: ``uint8(255) + uint8(1) == 0``. This situation is called
-an *overflow*. It occurs when an operation is performed that requires a fixed size variable
-to store a number (or piece of data) that is outside the range of the variable's data type.
-An *underflow* is the converse situation: ``uint8(0) - uint8(1) == 255``.
+They resemble integers when the values are small, but cannot represent arbitrarily large numbers.
+
+The following code causes an overflow because the result of the addition is too large
+to be stored in the type ``uint8``:
+
+::
+
+  uint8 x = 255;
+  uint8 y = 1;
+  return x + y;
+
+Solidity has two modes in which it deals with these overflows: Checked and non-checked or "wrapping" mode.
+
+The default checked mode will detect overflows and cause a failing assertion. You can disable this check
+using ``unchecked { ... }``, causing the overflow to be silently ignored. The above code would return
+``0`` if wrapped in ``unchecked { ... }``.
+
+Even in checked mode, please do not rely on these checks, since there are situations where an overflow
+can lead to a contract being stuck.
```

Comment:
What do you mean by "that"? Yes, SafeMath should not be needed anymore.

Reply:
> Even in checked mode, please do not rely on these checks, since there are situations where an overflow
can lead to a contract being stuck.

Can you give an example of this situation?

I also think it's a good idea to write somewhere that safe math libraries are redundant after 0.8.0.